source/numerics/data_out_resample.cc

```diff
@@ -0,0 +1,192 @@
+
+      cell->get_dof_indices(dof_indices);
+
+      for (unsigned int i = 0; i < dof_indices.size(); ++i)
+        points_all.emplace_back(partitioner->global_to_local(dof_indices[i]),
+                                fe_values.quadrature_point(i));
+    }
+
+  std::sort(points_all.begin(),
+            points_all.end(),
+            [](const auto &a, const auto &b) { return a.first < b.first; });
+  points_all.erase(std::unique(points_all.begin(),
+                               points_all.end(),
+                               [](const auto &a, const auto &b) {
+                                 return a.first == b.first;
+                               }),
+                   points_all.end());
+
+  std::vector<Point<spacedim>> points;
+
+  for (const auto &i : points_all)
+    {
+      point_to_local_vector_indices.push_back(i.first);
+      points.push_back(i.second);
+    }
+
+  rpe.reinit(points, *this->triangulation, *this->mapping);
+}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::build_patches(
+  const Mapping<dim, spacedim> &                                mapping,
+  const unsigned int                                            n_subdivisions,
+  const typename DataOut<patch_dim, spacedim>::CurvedCellRegion curved_region)
+{
+  this->update_mapping(mapping, n_subdivisions);
+  this->build_patches(curved_region);
+}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::build_patches(
+  const typename DataOut<patch_dim, spacedim>::CurvedCellRegion curved_region)
+{
+  patch_data_out.clear();
+
+  if (rpe.is_ready() == false)
+    update_mapping(*this->mapping, patch_dof_handler.get_fe().degree);
+
+  std::vector<std::shared_ptr<LinearAlgebra::distributed::Vector<double>>>
+    vectors;
+
+  patch_data_out.attach_dof_handler(patch_dof_handler);
+
+  unsigned int counter = 0;
+
+  for (const auto &data : this->dof_data)
+    {
+      const auto data_ptr = dynamic_cast<
+        internal::DataOutImplementation::
+          DataEntry<dim, spacedim, LinearAlgebra::distributed::Vector<double>>
+            *>(data.get());
+
+      // TODO: enable different vectors
+      Assert(data_ptr, ExcNotImplemented());
+
+      const auto &dh = *data_ptr->dof_handler;
+
+      // TODO: enable more components
+      AssertDimension(dh.get_fe_collection().n_components(), 1);
+
+      const auto values =
+        VectorTools::point_values<1>(rpe, dh, data_ptr->vector);
+
+      vectors.emplace_back(
+        std::make_shared<LinearAlgebra::distributed::Vector<double>>(
+          partitioner));
+
+      for (unsigned int j = 0; j < values.size(); ++j)
+        vectors.back()->local_element(point_to_local_vector_indices[j]) =
+          values[j];
+
+      vectors.back()->set_ghost_state(true);
+
+      patch_data_out.add_data_vector(
+        *vectors.back(),
+        std::string("temp_" + std::to_string(counter++)),
```

Comment:
Why not let the user provide the names (and, for that matter, component interpretation) of vectors? It seems unfriendly to just call all output variables `temp_*`.

Separately, this is a really difficult to find place where the counter is incremented. You're very much asking for mistakes if someone uses the counter below if the code below is ever changed. Please move the increment of `counter` to the very last line of the loop.

Reply:
> Why not let the user provide the names (and, for that matter, component interpretation) of vectors? It seems unfriendly to just call all output variables temp_*.

The variable names are not relevant here. It is actually not used, since the actual output of the data is not done by `patch_data_out`.

Reply:
Ah, I see. In that case, please add a comment to that effect.